quickwit-cli/src/index.rs

```diff
@@ -799,7 +801,10 @@ pub async fn start_statistics_reporting_loop(
         report_interval.tick().await;
         // Try to receive with a timeout of 1 second.
         // 1 second is also the frequency at which we update statistic in the console
-        let observation = pipeline_handle.observe().await;
+        let observations = client
+            .observe_pipeline(index_id.to_string(), Some(source_id.to_string()))
```

Comment:
Since there's one indexing pipeline per source, the `observe_pipeline` API becomes quite quirky. If no source ID is specified, it returns a map from (`index_id`, `source_id)` -> `Observation<IndexingStatistics>` for each source defined for the index. Otherwise, it returns a map with one value if the source exists.

Reply:
We can also split the API in two:
```
observe_pipeline(index_id, source_id) -> Option<IndexingStatistics>
observe_pipelines(index_id) -> HashMap<String, IndexingStatistics>
```

Same for the spawn API: `spawn_pipeline`, `spawn_pipeline_with_source`.